src/views/EditorView/Editor/Editor.tsx

```diff
@@ -38,91 +28,62 @@ interface IProps {
-            ImageLoadManager.addAndRun(this.loadImage(this.props.imageData));
-        }
-        if (prevProps.activeLabelType !== this.props.activeLabelType) {
-            this.swapSupportRenderingEngine(this.props.activeLabelType)
-        }
-        this.resizeCanvas(this.props.size);
-        this.calculateImageRect(this.state.image);
-        this.fullCanvasRender();
+    public componentDidUpdate(prevProps: Readonly<IProps>, prevState: Readonly<{}>, snapshot?: any): void {
+        const {imageData, activeLabelType} = this.props;
+
+        prevProps.imageData.id !== imageData.id && ImageLoadManager.addAndRun(this.loadImage(imageData));
+        prevProps.activeLabelType !== activeLabelType && EditorActions.swapSupportRenderingEngine(activeLabelType);
+
+        this.updateModelAndRender()
     }
 
     // =================================================================================================================
     // EVENT HANDLERS
     // =================================================================================================================
 
-    private update = (event: MouseEvent) => {
-        const editorData: EditorData = this.buildEditorData(event);
-        this.mousePositionOnCanvas = CanvasUtil.getMousePositionOnCanvasFromEvent(event, this.canvas);
-        this.primaryRenderingEngine.update(editorData);
-        this.supportRenderingEngine && this.supportRenderingEngine.update(editorData);
-        !this.props.activePopupType && this.updateMousePositionIndicator(event);
-        this.fullCanvasRender();
-    };
+    private mountEventListeners() {
+        window.addEventListener(EventType.MOUSE_MOVE, this.update);
+        window.addEventListener(EventType.MOUSE_UP, this.update);
+        EditorModel.canvas.addEventListener(EventType.MOUSE_DOWN, this.update);
+    }
 
-    private onMouseDown = (event: MouseEvent) => {
-        this.register();
-        this.update(event);
-    };
+    private unmountEventListeners() {
+        window.removeEventListener(EventType.MOUSE_MOVE, this.update);
+        window.removeEventListener(EventType.MOUSE_UP, this.update);
+        EditorModel.canvas.removeEventListener(EventType.MOUSE_DOWN, this.update);
+    }
 
     // =================================================================================================================
     // LOAD IMAGE
     // =================================================================================================================
 
     private loadImage = async (imageData: ImageData): Promise<any> => {
         if (imageData.loadStatus) {
-            this.setState({image: ImageRepository.getById(imageData.id)})
+            EditorModel.image = ImageRepository.getById(imageData.id);
+            this.updateModelAndRender()
         }
         else {
-            if (!this.isLoading) {
-                this.isLoading = true;
+            if (!EditorModel.isLoading) {
+                EditorModel.isLoading = true;
```

Comment:
generalnie każdą zmianę w modeulu powinieneś raczej robić przez akcję. czyli tutaj (i wyżej też, i w każdym innym miejscu)
robi się to po to żeby po prostu była spójność. żebyś mógł logować takie rzeczy itd... jest dużo więcej zalet, chociażby refaktoring

Reply:
więc pozamykaj w akcje wszystko co ma modyfikacje na modelu